src/odemis/acq/stream/_sync.py

```diff
@@ -1806,6 +1994,17 @@ def _runAcquisition(self, future):
                         raise CancelledError()
                     s._dataflow.unsubscribe(self._subscribers[i])
 
+                px_x, px_y = divmod(spots_sum, rep[0])
+                px_idx = (px_x, px_y)
+
+                for i, das in enumerate(self._acq_data):
+                    self._assembleLiveData2D(i, das[-1], px_idx, rep, 0)
+
+                # Activate _updateImage thread
```

Comment:
That's pretty much the exact description of what `shouldUpdateImage()` does. Indicate *why* now and what are the concequences. Maybe something like "Update the live image of each stream".
That said, how does it work for the other streams than the first one? I don't see where is the code...